### ISSUE TITLE
scrollbar horizontal/vertical up/down arrow scroll adjustment

### ISSUE THREAD

**Issue by ghost** [geany/geany#1904]
I have not been able to find a setting for this. I hope I've just missed it. On a new install of Fedora 28 Mate with geany 1.33 I'm finding that clicking on the horizontal or vertical, up or down arrows on the scrollbar is virtually useless. A single click doesn't appear to move at all. It's only apparent that there is any movement when you hold down on the mouse and then it's at a very slow crawl, vertically about 2 lines a second, horizontally less than a character per second. I can't imagine that this is the default expected behavior. Is there a setting or a fix for this?

**Comment by ronbanks** [comment]
Thanks, I'll have to look into the Scintilla angle on this but it's not anything to do with the theme. All other applications behave normally as expected and it's only the geany editor that behaves like this. Still looking for a solution.

**Comment by ronbanks** [comment]
Yeah, it doesn't look like there's not a setting in geany.conf to control this behaviour. This is a bug then.

**Comment by elextr** [comment]
There is not a setting in Geany.conf since (as shown in the link I posted) there is not a setting in the Scintilla API that Geany could set to control the look feel and speed of the scrollbar arrows.

The reason I say it follows the theme is that on my Mint Cinnamon there are no arrows, but that may just be Mate's GTK2 vs Cinnamon's GTK3.

**Comment by b4n** [comment]
Seems like it could be related to smooth scrolling, and Scintilla doesn't support this.  However, it sounds a little weird as normally the scrollbar arrows should be controlling the adjustment directly, just like when you actually move the scrollbar's slider.
@ronbanks could you tell us which theme you are using?  I don't have scrollbars, and it could make it easier not to have to search for such a theme :)

**Comment by ronbanks** [comment]
I didn't have a theme set so it was the default on Fedora 28. I switched themes several times to see if it made any difference and none of them changes this behavior. I've settled on the **TraditionalOk** theme now.

**Comment by toxpal** [comment]
I miss such a feature (to do proper scrolling) too. It's not related to theme because I have this issue on all themes.

**Comment by b4n** [comment]
Do you guys use Geany build with GTK2 or GTK3? You can check in help->debug messages

**Comment by ronbanks** [comment]
The only references that I see in the debug messages are:
07:37:57: Geany INFO		: GTK 3.22.30, GLib 2.56.1
07:37:58: Geany INFO		: Loaded GTK+ CSS theme '/usr/share/geany/geany.css'
07:37:58: Geany INFO		: Loaded GTK+ CSS theme '/usr/share/geany/geany-3.20.css'

**Comment by ronbanks** [comment]
Since inquiring about this issue, I've noticed the same scrolling on mysql workbench community 6.3 which looks like it uses GTK3 on linux.

**Comment by b4n** [comment]
Could be, I checked the difference in behavior between GTK2 and GTK3 with stock GTK scrolling and it's scrolling faster on GTK2 already.  There might be a weird change in the scollbars themselves and how they affect the adjustment that leads to this.  I'll see what we can do here.

**Comment by ronbanks** [comment]
upgraded mysql workbench community to latest and same scrolling problem but then I suppose it uses the same gtk I was hoping there had been a fix... I'll see if I can get hold of a downgraded version and test later.

**Comment by b4n** [comment]
Checking the scroll values we get from the scrollbar itself on GTK3, it's ridiculous: a single (short) click on one of the steppers leads to 5 steps of `0.5` each, for a total of `0.25`.  When the minimal step configured (and thus, expected) is `1`, it gives odd results.  You basically have to click *4 times* to get a single whole step.

On GTK2 we get a reasonable value of the configured step (`1`) for each click, leading to the expected behavior.

We probably should see if there's a good way to fix this, or report it to GTK.

**Comment by b4n** [comment]
@elextr I don't think so, see https://gitlab.gnome.org/GNOME/gtk/blob/gtk-3-24/gtk/gtkrange.c#L2961

**Comment by elextr** [comment]
Well, depends on how they set `step` (or the field they load it from I mean).

**Comment by codebrainz** [comment]
I just assumed it was an attempt at "smooth scrolling" in gtk3, to do like other toolkits.

**Comment by b4n** [comment]
@codebrainz giving `0.05` steps?  yeah it probably is, the problem is that the sum of the smallest scroll is smaller than the configured `step`.

**Comment by andy5995** [comment]
Operating system: Slackware-current (post 14.2)

I can move the horizontal scroll bar left-to-right in the editor pane. I can NOT drag the same scroll bar up or down to adjust the size of the status pane below it. (left-mouse click, hold, drag).

In the lower-left-hand pane (status, compiler, etc), I can NOT move the vertical scroll bar up or down. I can NOT adjust the size of the pane (left-mouse click, hold, drag).

If I click "compiler", the vertical scroll bar moves down a little as intended.

Using the mouse wheel in that pane does nothing.

No warnings or error show in the build output.

[config.log](https://github.com/geany/geany/files/2793584/config.log)

```
14:57:10: Geany INFO		: Using alternate configuration directory
14:57:10: Geany INFO		: Geany 1.35 (git >= bf5c9edd), en_US.UTF-8
14:57:10: Geany INFO		: GTK 3.24.4, GLib 2.58.3
14:57:10: Geany INFO		: System data dir: /home/andy/test-geany/share/geany
14:57:10: Geany INFO		: User config dir: /home/andy/test-geany/config
14:57:11: Geany INFO		: Loaded GTK+ CSS theme '/home/andy/test-geany/share/geany/geany.css'
14:57:11: Geany INFO		: Loaded GTK+ CSS theme '/home/andy/test-geany/share/geany/geany-3.20.css'
14:57:11: Geany INFO		: System plugin path: /home/andy/test-geany/lib/geany
14:57:11: Geany INFO		: Added filetype Clojure (61).
14:57:11: Geany INFO		: Added filetype Genie (62).
14:57:11: Geany INFO		: Added filetype Arduino (63).
14:57:11: Geany INFO		: Added filetype Scala (64).
14:57:11: Geany INFO		: Added filetype CUDA (65).
14:57:11: Geany INFO		: Added filetype JSON (66).
14:57:11: Geany INFO		: Added filetype Cython (67).
14:57:11: Geany INFO		: Added filetype Graphviz (68).
14:57:11: Geany INFO		: Loaded libvte from libvte-2.91.so
14:57:11: Geany INFO		: Loaded:   /home/andy/test-geany/lib/geany/debugger.so (Debugger)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/Makefile.am : Make (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/Makefile.in : Make (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/bst.c : C (UTF-8)
14:57:11: Geany INFO		: Loaded /home/andy/test-geany/share/geany/tags/std99.c.tags (C), 1618 symbol(s).
14:57:11: Geany INFO		: /home/andy/src/rmw/src/bst.h : C++ (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/config_rmw.c : C (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/config_rmw.h : C++ (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/gettext.h : C++ (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/messages_rmw.c : C (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/messages_rmw.h : C++ (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/purging_rmw.c : C (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/purging_rmw.h : C++ (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/restore_rmw.c : C (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/restore_rmw.h : C++ (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/rmw.c : C (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/rmw.h : C++ (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/strings_rmw.c : C (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/strings_rmw.h : C++ (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/trashinfo_rmw.c : C (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/trashinfo_rmw.h : C++ (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/trivial_rmw.c : C (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/trivial_rmw.h : C++ (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/utils_rmw.c : C (UTF-8)
14:57:11: Geany INFO		: /home/andy/src/rmw/src/utils_rmw.h : C++ (UTF-8)
```

**Comment by elextr** [comment]
> I can move the horizontal scroll bar left-to-right in the editor pane. I can NOT drag the same scroll bar up or down to adjust the size of the status pane below it. (left-mouse click, hold, drag).

the adjuster for the panes is below the scrollbar, not the scrollbar itself, its looks are theme dependent, its basically invisible on my theme

> In the lower-left-hand pane (status, compiler, etc), I can NOT move the vertical scroll bar up or down. I can NOT adjust the size of the pane (left-mouse click, hold, drag).

Thats a standard GTK scrollbar, so thats a GTK issue. The size adjustment see above.

> If I click "compiler", the vertical scroll bar moves down a little as intended.

If you mean you switch to that tab, each tab has its own scroll position, so its expected that the scrollbar position will change as you swap tabs.

> Using the mouse wheel in that pane does nothing.

Does it scroll the symbols pane?

**Comment by andy5995** [comment]
> the adjuster for the panes is below the scrollbar, not the scrollbar itself, its looks are theme dependent, its basically invisible on my theme

Ok. On buster when I hover the mouse just below the scrollbar, I get a double vertical arrow. On Slackware that does not appear, but now I see that I can indeed resize by clicking below the scroll bar, same as on Buster, just w/o the double-arrow appearing.

> 
> > In the lower-left-hand pane (status, compiler, etc), I can NOT move the vertical scroll bar up or down. I can NOT adjust the size of the pane (left-mouse click, hold, drag).
> 
> Thats a standard GTK scrollbar, so thats a GTK issue. The size adjustment see above.
> 
> 
> Does it scroll the symbols pane?

I'll try that next time I boot to Slackware. Probably ok. I think part of my confusion was this minor bug related to the Debugger plugin I mentioned https://github.com/geany/geany-plugins/pull/791#issuecomment-457364769. (I couldn't scroll the lower-left pane at all when the debugger plugin was enabled)
